### PR TITLE
Feature/rework office controller

### PR DESCRIPTION
**Pull request by SolovyovDmitry**

### CODE REVIEW

File: src/OfficeService/Controllers/OfficeController.cs [hunk 1]
```diff
@@ -62,12 +64,29 @@ public class OfficeController : ControllerBase
...
-    public async Task<OperationResultResponse<bool>> EditAsync(
-      [FromServices] IEditOfficeCommand command,
+    public async Task<IActionResult> EditAsync(
       [FromQuery] Guid officeId,
-      [FromBody] JsonPatchDocument<EditOfficeRequest> request)
+      [FromBody] JsonPatchDocument<EditOfficePatch> patch,
+      CancellationToken ct)
     {
-      return await command.ExecuteAsync(officeId, request);
+      if (!await _accessValidator.HasRightsAsync(Rights.AddEditRemoveCompanies))
```
Comment (VLiamin) [inline src/OfficeService/Controllers/OfficeController.cs:57]: Is it good practice to do everything in the controller?

File: src/OfficeService/Controllers/OfficeController.cs [hunk 2]
```diff
@@ -62,12 +64,29 @@ public class OfficeController : ControllerBase
...
+      {
+        return StatusCode(403);
+      }
+
+      var validationResult = await _validator.ValidateAsync(patch, ct);
```
Comment (VLiamin) [inline src/OfficeService/Controllers/OfficeController.cs]: May be remove var?

File: src/OfficeService.Business/Commands/Office/Edit/EditOfficeRequest.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using MediatR;
+using Microsoft.AspNetCore.JsonPatch;
+
+namespace LT.DigitalOffice.OfficeService.Business.Commands.Office.Edit
+{
+  public class EditOfficeRequest : IRequest<bool>
```
Comment (VLiamin) [inline src/OfficeService.Business/Commands/Office/Edit/EditOfficeRequest.cs]: May be make this record?

File: src/OfficeService.Business/Commands/Office/Edit/EditOfficeHandler.cs [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+      _httpContextAccessor = httpContextAccessor;
+      _globalCache = globalCache;
+      _provider = provider;
+    }
+
+    public async Task<bool> Handle(EditOfficeRequest request, CancellationToken ct)
+    {
+      bool result = await EditOfficeAsync(request.OfficeId, Map(request.Patch), ct);
+
+      var isActiveOperation = request.Patch.Operations.FirstOrDefault(
```
Comment (VLiamin) [inline src/OfficeService.Business/Commands/Office/Edit/EditOfficeHandler.cs]: May be remove var?

File: src/OfficeService/Controllers/OfficeController.cs (hunk 1 above)
Comment (SolovyovDmitry) [reply src/OfficeService/Controllers/OfficeController.cs:57]: There is a problem with what to return from handler in case of failed authentication

**Review by ndoriomedov — APPROVED** [review]

**Review by VLiamin — APPROVED** [review]

**Pull request merged**